spec/unit/recipes/chefdk_spec.rb

```diff
@@ -86,7 +86,7 @@
         end
 
         it 'upgrades chefdk' do
-          expect(chef_run).to upgrade_chef_ingredient('chefdk')
+          expect(chef_run).to install_chef_ingredient('chefdk')
```

Comment:
So internally `chef_ingredient` now fetches artifacts with the `remote_file` resource and then installs the artifact with the `package` resource. In this configuration the real indicator of whether an upgrade will occur is tied directly to whether the desired version has changed. You can provide a version via the `version` attribute (say `0.12.0`). You can also provide a version of `latest` which allows `chef_ingredient` to resolve that logical version to a real version. If the version that `latest` resolves to changes between CCRs then an upgrade will be done no matter what action you provide.